activerecord/lib/active_record/core.rb

```diff
@@ -728,8 +729,17 @@ def strict_loading?
     #   user.strict_loading!(false) # => false
     #   user.comments
     #   => #<ActiveRecord::Associations::CollectionProxy>
-    def strict_loading!(value = true)
+    def strict_loading!(value = true, mode: :all)
+      unless [:all, :n_plus_one_only].include?(mode)
+        raise ArgumentError, "`mode` must be one of `:all`, `:n_plus_one_only`."
```

Comment:
Maybe following reads better.
```suggestion
        raise ArgumentError, "The :mode option must be one of [:all, :n_plus_one_only]."
```